src/molara/rendering/rendering.py

```diff
@@ -232,10 +234,14 @@ def draw_cylinders_from_to(
         _directions: list[list[floating]] = []
         _lengths: list[floating] = []
         _positions_middle: list[list[list[floating]]] = []
+        zero_approx = 1e-6
         for pos12 in positions:
             pos1, pos2 = pos12
+            _length = np.linalg.norm(pos2 - pos1)
+            if _length < zero_approx:
+                continue
             _directions.append((pos2 - pos1).tolist())
-            _lengths.append(np.linalg.norm(pos2 - pos1))
+            _lengths.append(_length)
             _positions_middle.append((0.5 * (pos1 + pos2)).tolist())
```

Comment:
Consider replacing lines 237-245 with:

```
pos1, pos2 = positions[:, 0], positions[:, 1]
lengths = np.linalg.norm(pos2 - pos1, axis=1)
valid = lengths > np.finfo(np.float32).eps
_directions = (pos2 - pos1)[valid].tolist()
_lengths = lengths[valid].tolist()
_positions_middle = (0.5 * (pos1 + pos2))[valid].tolist()
```

FYI, `np.finfo(np.float32).eps` equals to 1.1920929e-07